BeanHub.API/src/main/java/com/bean/api/services/RecipeService.java

```diff
@@ -61,84 +63,91 @@ public List<Recipe> getFilteredRecipesByUsername(String username) {
-        String recipeJpql = "SELECT r FROM Recipe r WHERE r.recipeId IN :recipeIds";
-        TypedQuery<Recipe> recipeQuery = entityManager.createQuery(recipeJpql, Recipe.class);
-        recipeQuery.setParameter("recipeIds", recipeIds);
-        List<Recipe> recipes = recipeQuery.getResultList();
-        recipes.sort((r1, r2) -> {
-            LocalDateTime date1 = r1.getDateAdded();
-            LocalDateTime date2 = r2.getDateAdded();
-            return comparator.compare(date1 != null ? date1.toEpochSecond(ZoneOffset.UTC) : null,
-                    date2 != null ? date2.toEpochSecond(ZoneOffset.UTC) : null);
-        });
-        return recipes;
+    //Resolved
+    @Transactional(readOnly = true)
+    private List<Recipe> getSortedRecipesNewest() {
+        String jpql = "SELECT r FROM Recipe r ORDER BY r.dateAdded DESC";;
+        TypedQuery<Recipe> query = entityManager.createQuery(jpql, Recipe.class);
+        return query.getResultList();
     }
 
-    private List<Recipe> getSortedRecipesByAverageRating(List<Long> recipeIds, Map<Long, Double> averageRatings,
-            String sort) {
-        switch (sort) {
-            case "4":
-                recipeIds.sort(Comparator.comparingDouble(averageRatings::get));
-                break;
-            case "3":
-                recipeIds.sort(Comparator.comparingDouble(averageRatings::get).reversed());
-                break;
-            default:
-                break;
-        }
-        // Fetch all recipes from the database
-        List<Recipe> allRecipes = entityManager.createQuery("SELECT r FROM Recipe r", Recipe.class).getResultList();
-        Map<Long, Recipe> recipeMap = allRecipes.stream()
-                .collect(Collectors.toMap(Recipe::getRecipeId, Function.identity()));
-        List<Recipe> sortedRecipes = new ArrayList<>();
-        for (Long recipeId : recipeIds) {
-            Recipe recipe = recipeMap.get(recipeId);
-            if (recipe != null) {
-                sortedRecipes.add(recipe);
-            }
-        }
-        return sortedRecipes;
+    //Resolved
+    @Transactional(readOnly = true)
+    private List<Recipe> getSortedRecipesOldest() {
+        String jpql = "SELECT r FROM Recipe r ORDER BY r.dateAdded ASC";;
+        TypedQuery<Recipe> query = entityManager.createQuery(jpql, Recipe.class);
+        return query.getResultList();
     }
 
+    //Resolved
+    @Transactional(readOnly = true)
+    public List<Recipe> getSortedRecipesHighest() {
+        String jpql = "SELECT r " +
+                      "FROM Recipe r " +
+                      "LEFT JOIN r.ratings rt " +
+                      "GROUP BY r " +
+                      "ORDER BY AVG(rt.ratingValue) DESC";
+    
+        TypedQuery<Recipe> query = entityManager.createQuery(jpql, Recipe.class);
+        return query.getResultList();
+    }
+
+    //Resolved
+    @Transactional(readOnly = true)
+    public List<Recipe> getSortedRecipesLowest() {
+        String jpql = "SELECT r " +
```

Comment:
Why is this query multiline as opposed to the rest? Sticking to one style(be it writing your query on one line or multiple) makes the code look a lot more consistent.